inst/shiny/server.R

```diff
@@ -218,6 +244,58 @@ shinyServer(function(input, output, session) {
       updateAllPdataInputs()
     }
   })
+ 
+  #Create UI for filtering samples based on annotations
+  output$FilterSamples <- renderUI({
+    annot = ''
+    if (!is.null(vals$counts)){
+      annot = colnames(pData(vals$counts))
+    }
+    selectInput("filteredSample", "Select Annotation:", c("none", annot))
+  })
+  observeEvent(input$filteredSample, {
+    output$filterSampleOptions <- renderUI({
+      if (input$filteredSample != "none")({
+        L = vector("list", 2)
+        L[[1]] = list(checkboxGroupInput("filterSampleChoices", 
+                                         label = "Select samples to keep",
+                                         choices = unique(pData(vals$counts)[,input$filteredSample])))
+        L[[2]] = list(actionButton("runFilterSample", "Filter"))
```

Comment:
I think it looks cleaner and is more intuitive to use this way, but if you prefer having all the filtering options tied to one filter button I could integrate my code with the rest of the filtering stuff.